Add ingress controller unit tests

sebrandon1 opened the pull request:
Note: It looks like the `Update()` is fundamentally broken because it does not appear you can modify IngressController objects once they are created.  I tried various things to try and get it to work.

I left my test in there but commented out until we decide what to do with the `Update()` function itself.

Other changes:
* Modified the `Delete()` function to set the object to nil and return to match other package(s) functionality.

Resolves: https://github.com/openshift-kni/eco-goinfra/issues/361

kononovn reviewed (changes requested):
Please check few comments above.

sebrandon1 commented on pkg/ingress/ingresscontroller.go:
```diff
@@ -16,20 +16,22 @@ import (
-	apiClient *clients.Settings
+	Object *operatorv1.IngressController
+	// api clients to interact with the cluster.
+	readerClient goclient.Reader
```
Done.

kononovn commented on pkg/ingress/ingress_test.go:
```diff
@@ -1 +1,241 @@
+		{
+			expectedError:       true,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "ingresscontroller object test not found in namespace test",
```
It looks like error is invalid. It has to be "The` ingresscontroller name is empty" no?

sebrandon1 replied:
Fixed.

kononovn commented on pkg/ingress/ingress_test.go:
```diff
@@ -1 +1,241 @@
+		{
+			expectedError:       true,
+			addToRuntimeObjects: true,
+			expectedErrorText:   "ingresscontroller object test not found in namespace test",
```
expected error has to be different

sebrandon1 replied:
Fixed.

kononovn reviewed (changes requested):
Please check 2 more comments.

sebrandon1 commented:
Updated for the comments left.  

I removed my commented out test for `Update()` but I'm pretty sure that function doesn't work still.  Do we want to maybe address that in another PR?

kononovn commented on pkg/ingress/ingresscontroller_test.go:
```diff
@@ -0,0 +1,241 @@
+		}
+	}
+}
+
```
Can you add tests for Get(),Exist(), Update - It's OK to have it commented.

sebrandon1 replied:
Done.

kononovn reviewed (changes requested):
Please add few more tests.

kononovn reviewed (approved):
/lgtm
